### PR TITLE
Fix issues in webSocket handshake

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java [hunk 1]
```diff
@@ -53,6 +55,20 @@
      */
     HandshakeFuture handshake(String[] subProtocols, boolean allowExtensions, int idleTimeout);
 
+    /**
+     * Complete the handshake of a given request. The connection will be timed out if the connection is idle for given
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java]: Here in the comments you have aligned everything into one line. But none of the other methods in the interface follow such convention. Can we follow one convention..

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java [hunk 2]
```diff
@@ -50,6 +51,8 @@
     private final HttpRequest httpRequest;
     private final WebSocketSourceHandler webSocketSourceHandler;
     private boolean isCancelled = false;
+    private boolean handshakeStarted = false;
+    private HttpRequest request;
 
     public WebSocketInitMessageImpl(ChannelHandlerContext ctx, HttpRequest httpRequest,
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java]: Can we change this name to DefaultWebSocketInitMessage

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java [hunk 3]
```diff
@@ -115,7 +134,8 @@ private HandshakeFuture handleHandshake(WebSocketServerHandshaker handshaker, in
         }
 
         try {
-            ChannelFuture future = handshaker.handshake(ctx.channel(), httpRequest);
+            ChannelFuture future = handshaker.handshake(ctx.channel(), httpRequest, headers,
+                                                        ctx.channel().newPromise());
             handshakeFuture.setChannelFuture(future);
             future.addListener(new GenericFutureListener<Future<? super Void>>() {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java]: Can we turn it to lamda ?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/WebSocketClientConnectorImpl.java [hunk 4]
```diff
@@ -34,21 +34,19 @@
 
     private final String remoteUrl;
     private final String subProtocols;
-    private final String target;
     private final int idleTimeout;
     private final Map<String, String> customHeaders;
 
     public WebSocketClientConnectorImpl(WsClientConnectorConfig clientConnectorConfig) {
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contractimpl/websocket/WebSocketClientConnectorImpl.java]: Can we change the name to DefaultWebSocketClientConnector

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketServerHandshakeHandler.java [hunk 5]
```diff
@@ -61,7 +66,31 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                     Constants.WEBSOCKET_UPGRADE.equalsIgnoreCase(headers.get(HttpHeaderNames.UPGRADE))) {
                 log.debug("Upgrading the connection from Http to WebSocket for " +
                                   "channel : " + ctx.channel());
-                handleWebSocketHandshake(httpRequest, ctx);
+                ChannelPipeline pipeline = ctx.pipeline();
```
Comment (shafreenAnfar) [inline components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketServerHandshakeHandler.java]: Can't move all this to the method we originally had ?

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java (hunk 1 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java]: I am not sure I understand the convention. It follows the same convention as the other methods as far as I can see.

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketServerHandshakeHandler.java (hunk 5 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/listener/WebSocketServerHandshakeHandler.java]: I am calling that method inside the handler this handler

File: components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java (hunk 1 above)
Comment (riyafa) [reply components/org.wso2.transport.http.netty/src/main/java/org/wso2/transport/http/netty/contract/websocket/WebSocketInitMessage.java]: Fixed

**Pull request merged**